### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by noahcooper**

### CODE REVIEW

**Review by xriter — CHANGES_REQUESTED** [review]
There are two bugs in _facebook-native.js_ 

1: in `exports.setApplicationId` the called function should be `setApplicationId` (now it states `getApplicationId`.

2: in `exports.setApplicationName` the called function should be `setApplicationName` (now it states `getApplicationName`.

File: www/facebook-native.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
 var exec = require('cordova/exec')
 
+exports.getApplicationId = function (s, f) {
+  exec(s, f, 'FacebookConnectPlugin', 'getApplicationId', [])
+}
+
+exports.setApplicationId = function (appId, s, f) {
+  exec(s, f, 'FacebookConnectPlugin', 'getApplicationId', [appId])
```
Comment (xriter) [inline www/facebook-native.js:8]:
```suggestion
  exec(s, f, 'FacebookConnectPlugin', 'setApplicationId', [appId])
```

File: www/facebook-native.js [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+}
+
+exports.getApplicationName = function (s, f) {
+  exec(s, f, 'FacebookConnectPlugin', 'getApplicationName', [])
+}
+
+exports.setApplicationName = function (appName, s, f) {
+  exec(s, f, 'FacebookConnectPlugin', 'getApplicationName', [appName])
```
Comment (xriter) [inline www/facebook-native.js:16]:
```suggestion
  exec(s, f, 'FacebookConnectPlugin', 'setApplicationName', [appName])
```

**Review by xriter — CHANGES_REQUESTED** [review]

**Comment by xriter** [comment]
Experienced during iOS testing:
When switching appID with `setApplicationId()` one might get an error saying that the url 'fbXXXXXXXXXXXXnull' doesn't exist in the CFBundleURLSchemes (where XXXXXXXX is the appID). These aren't added to the info.plist during plugin installation. Only for the original appID.

Maybe some documentation in the readme could state that one can resolve this, by adding something like the following lines to their config.xml:
`
<platform name="ios">
 <config-file parent="CFBundleURLTypes" target="*-Info.plist">
            <array>
                <dict>
                    <key>CFBundleURLSchemes</key>
                    <array>
                        <string>fbXXXXappID1XXXXnull</string>
                    </array>
                </dict>
                <dict>
                    <key>CFBundleURLSchemes</key>
                    <array>
                        <string>fbXXXXappID2XXXXnull</string>
                    </array>
                </dict>
                <dict>
                    <key>CFBundleURLSchemes</key>
                    <array>
                        <string>fbXXXXappID3XXXXnull</string>
                    </array>
                </dict>
            </array>
        </config-file>
</platform>
`

(Haven't tested this in Android yet).

**Comment by xriter** [comment]
Getting the following error while building:
"
Thread 1: "As of v9.0, you must initialize the SDK prior to calling any methods or setting any properties. You can do this by calling `FBSDKApplicationDelegate`'s `application:didFinishLaunchingWithOptions:` method. Learn more: https://developers.facebook.com/docs/ios/getting-startedIf no `UIApplication` is available you can use `FBSDKApplicationDelegate`'s `initializeSDK` method."
"
(in main.m on line 32:
 `int retVal = UIApplicationMain(argc, argv, nil, @"AppDelegate");`
)

Not sure what causes this or how to fix it.
The previous release doesn't have this problem.

**Comment by noahcooper** [comment]
@xriter thanks for your feedback. This release is still a work in progress, but should be finalized tomorrow.

**Pull request merged**